### PR TITLE
EBE4816 - Refresh deployment policies combo when creating new Slave Configuration 

### PR DESCRIPTION
**Pull request by gsanchezu**
- Fix NullPointerException when Cloud does not exist yet (not available in the ClientCache)
- Improve combo behaviour: select first option by default, change message when there are no compatible policies, and if there is only one policy select it by default.

@EfrenRey, 👍?

### CODE REVIEW

**Comment by allyourbotsarebelongtous** [comment]
Deployed ElasticBox instance is available at https://10.166.246.82 and Jenkins instance at https://10.166.246.74

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See https://jenkins.elasticbox.com/job/ebx_pull_request_jenkins_plugin/456/ for more details.

File: src/main/java/com/elasticbox/jenkins/SlaveConfiguration.java [hunk 1]
```diff
@@ -261,25 +261,38 @@ public ListBoxModel doFillBoxVersionItems(@RelativePath("..") @QueryParameter St
...
                                                 @QueryParameter String workspace,
                                                 @QueryParameter String box) {
 
-            ListBoxModel profiles = getEmptyListBoxModel("--Please choose policy box--", "");
             if (anyOfThemIsBlank(endpointUrl, workspace, box, token)) {
                 return getEmptyListBoxModel();
             }
 
+            Client client = (name != null && name.length() > 0)
+                    ? ClientCache.getClient(name) : new Client(endpointUrl, token);
```
Comment (EfrenRey) [inline src/main/java/com/elasticbox/jenkins/SlaveConfiguration.java]: Do you need check that the cached client has the same ```endpointUrl``` that passed by param?

File: src/main/java/com/elasticbox/jenkins/SlaveConfiguration.java [hunk 2]
```diff
@@ -261,25 +261,38 @@ public ListBoxModel doFillBoxVersionItems(@RelativePath("..") @QueryParameter St
             return DescriptorHelper.getBoxVersions(createClient(endpointUrl, token), workspace, box);
         }
 
-        public ListBoxModel doFillProfileItems(@RelativePath("..") @QueryParameter String endpointUrl,
+        public ListBoxModel doFillProfileItems(@RelativePath("..") @QueryParameter String name,
```
Comment (EfrenRey) [inline src/main/java/com/elasticbox/jenkins/SlaveConfiguration.java]: Where is called this method? Are updated with the new first parameter?

Comment (gsanchezu) [inline src/main/java/com/elasticbox/jenkins/SlaveConfiguration.java]: It is called by Jenkins automatically when the combo needs to be refreshed, and with the annotation we are telling Jenkins where to get the param from (a form field), so there is no calling method to be updated.

File: src/main/java/com/elasticbox/jenkins/SlaveConfiguration.java (hunk 1 above)
Comment (gsanchezu) [inline src/main/java/com/elasticbox/jenkins/SlaveConfiguration.java]: Hmmm, yes, maybe if the client is already cached but the user has changed the URL and not saved yet, we must check the `endpointUrl`... I changed the code, but in fact it affected most of the methods, so I changed all of them

**Comment by allyourbotsarebelongtous** [comment]
Deployed ElasticBox instance is available at https://10.166.246.82 and Jenkins instance at https://10.166.246.74

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See https://jenkins.elasticbox.com/job/ebx_pull_request_jenkins_plugin/457/ for more details.

File: src/main/java/com/elasticbox/jenkins/SlaveConfiguration.java (hunk 2 above)
Comment (EfrenRey) [inline src/main/java/com/elasticbox/jenkins/SlaveConfiguration.java]: 👍

**Comment by allyourbotsarebelongtous** [comment]
The following instances are being terminated: https://ops.elasticbox.com/services/instances/i-7o5yq5, https://ops.elasticbox.com/services/instances/i-cq3drh

**Pull request merged**